File: files/CH-068.R
library(tidyverse)

draw_triangle = function(n) {
  if (n %% 2 == 0) {
    return("Not Possible")
  } else {
    x = ceiling(n / 2)
    seq = seq(1, x)
    
    mat = matrix("", n, x)
    for (i in 1:x) {
      mat[i, 1:seq[i]] = "*"
    }
    for (i in 1:(x - 1)) {
      mat[n - i + 1, ] = mat[i, ]
    }
    
    print(mat)
  }
}


draw_triangle(7)

File: files/000-099/068/CH-068.py
import numpy as np

def draw_triangle(n):
    if n % 2 == 0:
        print("Not Possible")
    else:
        x = (n + 1) // 2
        seq = list(range(1, x + 1))       
        mat = np.full((n, x), "", dtype=object)
        for i in range(x):
            mat[i, :seq[i]] = "*"
        for i in range(1, x):
            mat[n - i, :] = mat[i - 1, :]
        print(mat)

draw_triangle(7)